app/internal/wm/GioView.java

```diff
@@ -242,13 +245,20 @@ private void dispatchMotionEvent(MotionEvent event) {
 		}
 	}
 
-	@Override public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
-		return new InputConnection(this);
+	@Override public InputConnection onCreateInputConnection(EditorInfo editor) {
+        editor.inputType = this.keyboardMode;
+        editor.imeOptions = EditorInfo.IME_FLAG_NO_FULLSCREEN | EditorInfo.IME_FLAG_NO_EXTRACT_UI;
+        return new InputConnection(this);
 	}
 
-	void showTextInput() {
-		GioView.this.requestFocus();
-		imm.showSoftInput(GioView.this, 0);
+	void showTextInput(int mode) {
+        this.keyboardMode = mode;
+        if (this.lastKeyboardMode != mode) {
+            imm.restartInput(this);
```

Comment:
_I didn't find anyway to change the type of the keyboard without restart it. The restart calls the `onCreateInputConnection` which sets the keyboard mode._